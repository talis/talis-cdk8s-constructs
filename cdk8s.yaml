language: typescript
imports:
  - k8s@1.30.0
