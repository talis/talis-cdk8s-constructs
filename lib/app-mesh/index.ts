export * from "./ingress-gateway-nlb";
export * from "./app-service";
