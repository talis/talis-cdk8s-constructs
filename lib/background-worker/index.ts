export * from "./background-worker-props";
export * from "./background-worker";
