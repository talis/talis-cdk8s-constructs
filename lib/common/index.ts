export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
