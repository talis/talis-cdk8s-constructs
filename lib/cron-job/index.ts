export * from "./cron-job-props";
export * from "./cron-job";
