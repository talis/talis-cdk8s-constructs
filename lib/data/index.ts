export * from "./config-map";
export * from "./secret";
export * from "./factory";
