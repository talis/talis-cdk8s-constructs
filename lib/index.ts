export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
