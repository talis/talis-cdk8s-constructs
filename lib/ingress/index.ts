export * from "./ingress-props";
export * from "./ingress";
