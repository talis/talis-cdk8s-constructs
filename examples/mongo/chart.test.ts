import { MongoChart } from "./chart";
import { Testing } from "cdk8s";
import { TalisShortRegion, TalisDeploymentEnvironment } from "../../lib";

describe("Mongo example", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("Snapshot", () => {
    const app = Testing.app();
    const chart = new MongoChart(app, {
      environment: TalisDeploymentEnvironment.DEVELOPMENT,
      region: TalisShortRegion.LOCAL,
      watermark: "test",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
