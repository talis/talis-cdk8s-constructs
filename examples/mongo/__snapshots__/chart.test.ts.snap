// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mongo example Snapshot 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": Object {
      "labels": Object {
        "app": "example-mongo-app",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "example-mongo-app-development-local",
      },
      "name": "example-mongo-app-test",
      "namespace": "example-mongo-app-test",
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "labels": Object {
        "app": "example-mongo-app",
        "environment": "development",
        "instance": "redis-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "3.2.8",
        "role": "mongo",
        "service": "example-mongo-app-development-local",
      },
      "name": "redis-example",
      "namespace": "example-mongo-app-test",
    },
    "spec": Object {
      "clusterIP": "None",
      "ports": Array [
        Object {
          "port": 27107,
          "protocol": "TCP",
        },
      ],
      "selector": Object {
        "app": "example-mongo-app",
        "instance": "redis-example",
        "role": "mongo",
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": Object {
      "labels": Object {
        "app": "example-mongo-app",
        "environment": "development",
        "instance": "redis-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "3.2.8",
        "role": "mongo",
        "service": "example-mongo-app-development-local",
      },
      "name": "redis-example-sts",
      "namespace": "example-mongo-app-test",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "example-mongo-app",
          "instance": "redis-example",
          "role": "mongo",
        },
      },
      "serviceName": "redis-example",
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "example-mongo-app",
            "instance": "redis-example",
            "role": "mongo",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "args": Array [
                "--smallfiles",
                "--storageEngine",
                "mmapv1",
              ],
              "image": "mongo:3.2.8",
              "name": "mongo",
              "ports": Array [
                Object {
                  "containerPort": 27017,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "100m",
                  "memory": "500Mi",
                },
              },
              "volumeMounts": Array [
                Object {
                  "mountPath": "/data/db",
                  "name": "mongo-data",
                },
              ],
            },
          ],
        },
      },
      "volumeClaimTemplates": Array [
        Object {
          "metadata": Object {
            "name": "mongo-data",
          },
          "spec": Object {
            "resources": Object {
              "requests": Object {
                "storage": "20Gi",
              },
            },
            "storageClassName": "general-purpose-delete",
          },
        },
      ],
    },
  },
]
`;
